develop/smart-contracts/overview.md

```diff
@@ -1,4 +1,145 @@
+
+At a high level, [Frontier](https://github.com/polkadot-evm/frontier){target=\_blank} is composed of three main components:
+
+- [**Ethereum Client**](https://github.com/polkadot-evm/frontier/tree/master/client){target=\_blank} - an Ethereum JSON RPC compliant client that allows any request coming from an Ethereum tool, such as [Remix](https://remix.ethereum.org/){target=\_blank}, [Hardhat](https://hardhat.org/){target=\_blank} or [Foundry](https://getfoundry.sh/), to be admitted by the network
+- [**Pallet Ethereum**](https://docs.rs/pallet-ethereum/latest/pallet_ethereum/){target=\_blank} - a block emulation and Ethereum transaction validation layer that works jointly with the Ethereum client to ensure compatibility with Ethereum tools
+- [**Pallet EVM**](https://docs.rs/pallet-evm/latest/pallet_evm/){target=\_blank} - access layer to the [Rust-based EVM](https://github.com/rust-ethereum/evm){target=\_blank}, enabling the execution of EVM smart contract logic natively
+
+Broadly speaking, in this configuration, an EVM transaction follows the path presented in the diagram below:
+
+``` mermaid
+flowchart TD
+    A[Users and Devs] -->|Send Tx| B[Frontier RPC Ext]
+    subgraph C[Pallet Ethereum]
+        D[Validate Tx]
+        E[Send<br/>Valid Tx]    
+    end
+    B -->|Interacts with| C
+    D --> E
+    subgraph F[Pallet EVM]
+        G[Rust EVM]
+    end
+    I[(Current EVM<br/>Emulated State)]
+
+    H[Smart Contract<br/>Solidity, Vyper...] <-->|Compiled to EVM<br/>Bytecode| F
+
+    C --> F
+    I --> F
+    F --> J[(New Ethereum<br/>Emulated State)]
+    F --> K[Execution Logs]
+
+    style C fill:#ffffff,stroke:#000000,stroke-width:1px
+    style F fill:#ffffff,stroke:#000000,stroke-width:1px
+```
+
+Although it seems complex, users and developers are abstracted of that complexity, and tools can easily interact with the parachain as they would with any other EVM-compatible environment.
+
+The Rust EVM is capable of executing regular [EVM bytecode](https://www.ethervm.io/){target=\_blank}. Consequently, any language that compiles to EVM bytecode can be used to create programs that the parachain can execute.
+
+ou can find more information on deploying EVM smart contracts to [Polkadot's native smart contract platform](TODO:update-path){target=\_blank}, or any of [the ecosystem parachains](TODO:update-path){target=\_blank}.
+
+### Contracts Pallet
+
+The [`pallet_contracts`](https://docs.rs/pallet-contracts/latest/pallet_contracts/index.html#contracts-pallet){target=\_blank} provides the execution environment for Wasm-based smart contracts. Consequently, any smart contract language that compiles to Wasm can be executed in a parachain that enables this module.
+
+At the time of writing there are two main languages that can be used for Wasm programs:
```

Comment:
Nitpick here, but `At the time of writing there are two main languages that can be used for Wasm programs:` is inaccurate because `Solang` is not a language.